providers/gpgpu/bin/gpu_passthrough.py

```diff
@@ -0,0 +1,576 @@
+    total_runtime_sec = 0.0
+    for i in range(run_count):
+        tic = time.time()
+        # XXX: If we package the test, this line needs to be updated
+        instance.run("./test", on_guest=True, check=True)
+        toc = time.time()
+        runtime_sec = toc - tic
+        total_runtime_sec += runtime_sec
+        logging.debug("Runtime #%d (sec): %f", i, runtime_sec)
+
+    avg_runtime_sec = total_runtime_sec / run_count
+    logging.info("Average runtime (sec): %f", avg_runtime_sec)
+    if avg_runtime_sec >= threshold_sec:
+        logging.error(
+            "Average runtime %fs greater than threshold %fs",
+            avg_runtime_sec,
+            threshold_sec,
+        )
+        raise SystemExit(1)
+
+
+def test_lxd_gpu(args):
+    """Tests GPU passthrough with a LXD container."""
+    logging.info("Executing LXD GPU passthrough test")
+
+    instance = LXD(args.template, args.rootfs)
+    try:
+        instance.init_lxd()
+        options = []
+        if args.vendor == "nvidia":
+            options = ["-c nvidia.runtime=true"]
+        instance.launch(options=options)
+        instance.add_device("gpu", "gpu", options=["pci={}".format(args.pci)])
+
+        logging.info("Waiting for network to be up")
+        time.sleep(20)
+
+        build_gpu_test(instance, args.vendor)
+        run_gpu_test(instance, args.count, args.threshold)
+    finally:
+        instance.cleanup()
+
+
+def test_lxdvm_gpu(args):
+    """Tests GPU passthrough with a LXD virtual machine."""
+    logging.info("Executing LXD VM GPU passthrough test")
+
+    instance = LXDVM(args.template, args.image)
+    try:
+        instance.init_lxd()
+        instance.launch(
+            options=["-d root,size=50GB", "-c security.secureboot=false"]
+        )
+
+        logging.info("Waiting for network to be up")
+        time.sleep(20)
+
+        # Add and configure GPU device
+        instance.add_device("gpu", "gpu", ["pci={}".format(args.pci)])
+        if args.vendor == "nvidia":
+            logging.debug("Installing ubuntu-drivers tool")
+            instance.run(
+                "apt-get -q install -y ubuntu-drivers-common",
+                on_guest=True,
+                check=True,
+            )
+            instance.run("ubuntu-drivers install", on_guest=True, check=True)
+        instance.restart()
+
+        logging.info("Waiting for network to be up")
+        time.sleep(20)
+
+        build_gpu_test(instance, args.vendor)
+        run_gpu_test(instance, args.count, args.threshold)
+    finally:
+        instance.cleanup()
+
+
+def parse_args():
+    """Parses command-line arguments."""
+    parser = argparse.ArgumentParser(description="GPU passthrough tests")
+    subparsers = parser.add_subparsers()
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        dest="log_level",
+        action="store_const",
+        default=logging.INFO,
+        const=logging.DEBUG,
+        help="Increase logging level",
+    )
```

Comment:
I'm not sure I see the value of this, why would we hide the output of most commands most of the times?

Reply:
I've added a few more `logging.info` messages to show what step the program is on.
I also increased the level for some error messages that used to be debug before. If you think I should do more, let me know.